config/resource.go

```diff
@@ -0,0 +1,19 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package config // import "go.opentelemetry.io/contrib/config"
+
+import (
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
+)
+
+func newResource(res *Resource) (*resource.Resource, error) {
+	if res == nil || res.Attributes == nil {
+		return resource.Default(), nil
+	}
+	return resource.Merge(resource.Default(),
+		resource.NewWithAttributes(*res.SchemaUrl,
+			semconv.ServiceName(*res.Attributes.ServiceName),
```

Comment:
Should it not be possible to set other (additional) attributes as well?
https://github.com/open-telemetry/opentelemetry-configuration/blob/62138088f545154d15aca4e74fe6f6ae93c94454/schema/resource.json#L19

Should we create an issue for it?

Reply:
Yes. I need to follow up on this, because the generated code doesn't support a map currently, see https://github.com/open-telemetry/opentelemetry-go-contrib/blob/044e09ac8bfe0d5a60510778017bec57413243d9/config/generated_config.go#L19-L22

Reply:
Opened https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4828